utils/taskcluster_downloader.py

```diff
@@ -64,14 +66,29 @@ def donwload_logs(group_id, output):
         os.makedirs(output, exist_ok=True)
         output_path = os.path.join(output, f"{task_obj_label}.log")
 
-        print(f"Downloading {output_path}")
-        with open(output_path, "wb") as f:
-            downloadArtifactToFile(
-                f,
-                taskId=task["status"]["taskId"],
-                name="public/build/train.log",
-                queueService=queue,
-            )
+        url = queue.buildUrl("getLatestArtifact", task_id, "public/logs/live.log")
+        resp = requests.get(url, stream=True, timeout=5)
```

Comment:
suggestion (non-blocking): Consider moving this and the other parts of log processing out to an asynch function.

Now that this loop is handling running tasks as well, downloading logs synchronously may be less than ideal. For example, if there's two tasks that are both running you will end up sitting in the body of the loop for hours or even days depending on which task is looked at first. Downloading async will ensure you fetch all live logs as the tasks run.

Reply:
I use streaming and kill the connection on timeout, so it takes 5 seconds to download any live log. I would like not to spend time on this script because it's a very temporary workaround until we download W&B which I hope will happen soon.